File: javascript/matrix/matrix.js
export class Matrix {
  constructor(input) {
    this._rows = input.split("\n").map(x => x.split(" ").map( y => Number(y)));
    this._columns = this._rows[0].map((col, i) => this._rows.map(row => row[i]));
  }
  get rows(){
    return this._rows;
  }
  get columns() {
    return this._columns;
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.rows = list(map(lambda x : list(map(lambda y: int(y), x.split(" "))) , matrix_string.split("\n")))
        self.columns = list(map(list, zip(*self.rows)))

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return self.columns[index - 1]
